CMakeLists.txt

```diff
@@ -37,6 +37,7 @@ option(SOLC_STATIC_STDLIBS "Link solc against static versions of libgcc and libs
 option(STRICT_Z3_VERSION "Use the latest version of Z3" ON)
 option(PEDANTIC "Enable extra warnings and pedantic build flags. Treat all warnings as errors." ON)
 option(PROFILE_OPTIMIZER_STEPS "Output performance metrics for the optimiser steps." OFF)
+option(BUILD_LIBS_ONLY "Only build Solidity libraries and discard command line tools." OFF)
```

Comment:
I don't think we really need kind of option. That's the functionality you get out of the box as a part of the build system. For example if you're only interested in `liblangutil` and `libsmtutil`, you can simply invoke `make` like this:

```bash
make langutil smtutil
```

Reply:
> I don't think we really need such an option. That's the functionality you get out of the box as a part of the underlying build system. For example if you're only interested in `liblangutil` and `libsmtutil`, you can simply invoke `make` like this:
> 
> ```shell
> make langutil smtutil
> ```

Your comments only make sense if you are working on the Solidity compiler directly. In my case, I'm using Solidity as a third-party library, so in my CMakeLists.txt, I have something like:
```
# Disable Solidity tests to speed up the build.
set(TESTS OFF)
# Disable Z3 SMT solver to avoid dependency issues.
set(USE_Z3 OFF)
message(STATUS "Start fetching Solidity libraries")
FetchContent_Declare(
  solidity
  GIT_REPOSITORY https://github.com/ethereum/solidity.git
  GIT_TAG origin/v0.8.21
)

add_library(my_downstream_app ${sources})

target_link_libraries(
    my_downstream_app
    PRIVATE
    solidity)
```

In this case, you cannot simply use `make` to choose the subtarget because they all belong to `solidity` IIUC.